File: app/src/main/java/com/waveneuro/ui/session/history/SessionListAdapter.java
package com.waveneuro.ui.session.history;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.core.content.ContextCompat;

import androidx.recyclerview.widget.RecyclerView;

import com.waveneuro.R;

import java.util.List;

public class SessionListAdapter extends RecyclerView.Adapter<SessionListAdapter.ViewHolder> {

    private List<Session> sessions;

    public static class ViewHolder extends RecyclerView.ViewHolder {
        private final TextView tvName;
        private final TextView tvRd;
        private final TextView tvSd;
        private final TextView tvStatus;

        public ViewHolder(View view) {
            super(view);

            tvName = view.findViewById(R.id.tv_name);
            tvRd = view.findViewById(R.id.tv_rd);
            tvSd = view.findViewById(R.id.tv_sd);
            tvStatus = view.findViewById(R.id.tv_status);
        }
    }

    public SessionListAdapter(List<Session> s) {
        sessions = s;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup viewGroup, int viewType) {
        View view = LayoutInflater.from(viewGroup.getContext())
                .inflate(R.layout.item_session, viewGroup, false);

        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ViewHolder viewHolder, final int position) {

        viewHolder.tvName.setText(sessions.get(position).name);
        viewHolder.tvRd.setText(sessions.get(position).rd==null?"n/a":sessions.get(position).rd);
        viewHolder.tvSd.setText(sessions.get(position).sd);

        Context context = viewHolder.itemView.getContext();

        String completed = context.getString(R.string.completed);
        String terminated = context.getString(R.string.terminated);
        viewHolder.tvStatus.setText(sessions.get(position).isCompleted?completed:terminated);

        int color = ContextCompat.getColor(context, sessions.get(position).isCompleted?R.color.aqua: R.color.gray_dim_dark);
        viewHolder.tvStatus.setTextColor(color);
    }

    public int getItemCount() {
        return sessions.size();
    }
}

File: app/src/main/java/com/waveneuro/ui/session/history/adapter/SessionListAdapter.kt
package com.waveneuro.ui.session.history.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import com.waveneuro.R
import com.waveneuro.databinding.ItemSessionBinding
import com.waveneuro.ui.model.Session

class SessionListAdapter(
    private val context: Context,
    private val sessions: List<Session>
) : RecyclerView.Adapter<SessionListAdapter.ViewHolder>() {

    inner class ViewHolder(
        private val binding: ItemSessionBinding,
    ) : RecyclerView.ViewHolder(binding.root) {

        fun bind() {
            val session = sessions[absoluteAdapterPosition]
            with(binding) {
                tvName.text = session.name
                tvRd.text = session.rd ?: "n/a"
                tvSd.text = session.sd
                val text = context.getString(
                    if (session.isCompleted) R.string.completed else R.string.terminated)
                val color = ContextCompat.getColor(context,
                    if (session.isCompleted) R.color.aqua else R.color.gray_dim_dark)
                tvStatus.text = text
                tvStatus.setTextColor(color)
            }
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder =
        ViewHolder(ItemSessionBinding.inflate(LayoutInflater.from(context), parent, false))

    override fun onBindViewHolder(holder: SessionListAdapter.ViewHolder, position: Int) {
        holder.bind()
    }

    override fun getItemCount(): Int = sessions.size

}